app/lib/events/web_request.rb

```diff
@@ -42,5 +43,17 @@ def query_to_kv_pairs(query_string)
         { 'key' => key, 'value' => Array.wrap(value) }
       end
     end
+
+    def anonymised_user_agent_and_ip(rack_request)
+      if rack_request.remote_ip.present?
+        anonymise(rack_request.user_agent.to_s + rack_request.remote_ip.to_s)
+      else
+        ''
```

Comment:
@misaka I don't code Ruby, but is '' the same as null?

Reply:
Ah, ya, I made the quick assumption that it essentially gets serialised to the same thing when sent over to BigQuery, but thinking about it now I don't think that's the case. Will fix, thanks!

Reply:
[`b0fa8ba` (#833)](https://github.com/DFE-Digital/find-teacher-training/pull/833/commits/b0fa8baa8990543eccfa077071f90a57a609f56d)